types/react-native-dotenv/index.d.ts

```diff
@@ -0,0 +1,13 @@
+// Type definitions for react-native-dotenv 0.2
+// Project: https://github.com/zetachang/react-native-dotenv
+// Definitions by: hmajid2301 <https://github.com/hmajid2301>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
+
+interface Env {
+    [name: string]: string;
+}
+
+declare const env: Env;
+
+export { env };
```

Comment:
Instead of this export, you should do `export = env;`.
This way the import can match what will exist at runtime.
Unfortunately, TS won't let you do what's presented in the docs `import { ANY_STRING_HERE } from "react-native-dotenv";` - there's no way to say a module has any possible export. But you will be able to do `import env = require("react-dotnet-env");` or `import env from "react-dotnet-env"` with the `--esModuleInterop` flag.